tests/aws/services/stepfunctions/v2/test_state/test_test_state_scenarios.py

```diff
@@ -0,0 +1,246 @@
+import json
+
+import pytest
+from localstack_snapshot.snapshots.transformer import RegexTransformer
+
+from localstack.aws.api.lambda_ import Runtime
+from localstack.aws.api.stepfunctions import InspectionLevel
+from localstack.testing.pytest import markers
+from localstack.utils.strings import short_uid
+from tests.aws.services.stepfunctions.templates.services.services_templates import (
+    ServicesTemplates as ST,
+)
+from tests.aws.services.stepfunctions.templates.test_state.test_state_templates import (
+    TestStateTemplate as TCT,
+)
+
+HELLO_WORLD_INPUT = json.dumps({"Value": "HelloWorld"})
+NESTED_DICT_INPUT = json.dumps(
+    {
+        "initialData": {"fieldFromInput": "value from input", "otherField": "other"},
+        "unrelatedData": {"someOtherField": 1234},
+    }
+)
+BASE_CHOICE_STATE_INPUT = json.dumps({"type": "Private", "value": 22})
+
+BASE_TEMPLATE_INPUT_BINDINGS: list[tuple[str, str]] = [
+    (TCT.BASE_PASS_STATE, HELLO_WORLD_INPUT),
+    (TCT.BASE_RESULT_PASS_STATE, HELLO_WORLD_INPUT),
+    (TCT.IO_PASS_STATE, NESTED_DICT_INPUT),
+    (TCT.IO_RESULT_PASS_STATE, NESTED_DICT_INPUT),
+    (TCT.BASE_FAIL_STATE, HELLO_WORLD_INPUT),
+    (TCT.BASE_SUCCEED_STATE, HELLO_WORLD_INPUT),
+    (TCT.BASE_CHOICE_STATE, BASE_CHOICE_STATE_INPUT),
+]
+IDS_BASE_TEMPLATE_INPUT_BINDINGS: list[str] = [
+    "BASE_PASS_STATE",
+    "BASE_RESULT_PASS_STATE",
+    "IO_PASS_STATE",
+    "IO_RESULT_PASS_STATE",
+    "BASE_FAIL_STATE",
+    "BASE_SUCCEED_STATE",
+    "BASE_CHOICE_STATE",
+]
+
+
+@markers.snapshot.skip_snapshot_verify(
+    paths=[
+        "$..loggingConfiguration",
+        "$..tracingConfiguration",
+        "$..SdkHttpMetadata",
+        "$..SdkResponseMetadata",
+    ]
+)
+class TestStateCaseScenarios:
+    @markers.aws.validated
+    @pytest.mark.parametrize(
+        "tct_template,execution_input",
+        BASE_TEMPLATE_INPUT_BINDINGS,
+        ids=IDS_BASE_TEMPLATE_INPUT_BINDINGS,
+    )
+    def test_base_inspection_level_info(
+        self,
+        stepfunctions_client_test_state,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        sfn_snapshot,
+        tct_template,
+        execution_input,
+    ):
+        sfn_role_arn = create_iam_role_for_sfn()
+
+        template = TCT.load_sfn_template(tct_template)
+        definition = json.dumps(template)
+
+        test_case_response = stepfunctions_client_test_state.test_state(
+            definition=definition,
+            roleArn=sfn_role_arn,
+            input=execution_input,
+            inspectionLevel=InspectionLevel.INFO,
```

Comment:
The inspection level could be extra parameter and pytest automatically covers all combinations (e.g., https://stackoverflow.com/a/72215438/6875981).
Downside: the snapshot skips would apply to everything (I don't think we support conditional snapshot skips based on params); it would be pretty easy in this case (INFO ✅ , others ❌ )

Reply:
I agree, the snapshot skips required do represent a downside, but I am also wondering if expanding the parameter size and combinations would yield returning benefits at this point. I say this as test ids will also need expanding (probably appending the inspection level) and may make it inconvenient, although feasible, to rerun individual tests in the future.
I will add a comment to investigate aggregating all `test_base_` into a single parametrised function

Reply:
definitely a very valid point 👍 good pragmatic solution 👍 👏